### PR TITLE
Add setup.py for package publishing

### LINKED ISSUE #4: `pip install chime`
Related to https://github.com/pennsignals/chime/issues/6

Per this [slack message](https://codeforphilly.slack.com/archives/C01080RBV3Q/p1584462431020400), 

>  i was able to compare the model to the Portugal reality (black datapoints: still not enough data but ok). Im still working on it. What I had to do first was to decouple the code from the web-app integration (it has a few dependencies). maybe the core can transformed into a module ( #chime-app maybe?)

> 
image

I am going to guess at some desired functionality / API design, please discuss / correct me in the comments:

* Should be able to `pip install chime` or something similar
* Should be able to `from chime import  sim_sir` and model `sir` in code (? Not sure)
* should be able to ...

### PR DESCRIPTION
**Pull request by nickcanz**
This PR adds a setup.py script so that `penn_chime` can be a published package that is `pip install`-able.

I tested on my local machine by running:

```
pip3 install .
```

And in a python console I'm able to import things like `penn_chime.models` and see the output of `pydoc3`

```
$ pydoc3 penn_chime.models
Help on module penn_chime.models in penn_chime:

NAME
    penn_chime.models - Models.

DESCRIPTION
    Changes affecting results or their presentation should also update
    parameters.py `change_date`, so users can see when results have last
    changed

CLASSES
    builtins.object
        SimSirModel

    class SimSirModel(builtins.object)
     |  SimSirModel(p: 'Parameters') -> 'SimSirModel'
```

I also moved the `cli.py` script into the `penn_chime` module so that it can be installed along with the package. After running the pip install, you can execute `penn_chime`

```
$ penn_chime -h
usage: penn_chime [-h] [--file FILE] [--prefix PREFIX]
                  [--current-hospitalized CURRENT_HOSPITALIZED]
                  [--doubling-time DOUBLING_TIME]
                  [--hospitalized-los HOSPITALIZED_LOS]
                  [--hospitalized-rate HOSPITALIZED_RATE] [--icu-los ICU_LOS]
                  [--icu-rate ICU_RATE] [--known-infected KNOWN_INFECTED]
                  [--market_share MARKET_SHARE] [--n-days N_DAYS]
                  [--relative-contact-rate RELATIVE_CONTACT_RATE]
                  [--susceptible SUSCEPTIBLE]
                  [--ventilated-los VENTILATED_LOS]
                  [--ventilated-rate VENTILATED_RATE]

CHIME

optional arguments:
  -h, --help            show this help message and exit
  --file FILE
  --prefix PREFIX
  --current-hospitalized CURRENT_HOSPITALIZED
  --doubling-time DOUBLING_TIME
  --hospitalized-los HOSPITALIZED_LOS
  --hospitalized-rate HOSPITALIZED_RATE
  --icu-los ICU_LOS
  --icu-rate ICU_RATE
  --known-infected KNOWN_INFECTED
  --market_share MARKET_SHARE
  --n-days N_DAYS
  --relative-contact-rate RELATIVE_CONTACT_RATE
  --susceptible SUSCEPTIBLE
  --ventilated-los VENTILATED_LOS
  --ventilated-rate VENTILATED_RATE
```

## Still to-do

This PR just adds the scripting to create the package. Running `python3 setup.py sdist bdist_wheel` in the project root will create the files in a `dist/` directory.

To complete this, someone will still need to upload the package to pypi, I _believe_ this should be accomplished by following the packaging guide: https://packaging.python.org/tutorials/packaging-projects/#uploading-the-distribution-archives

Since I'm not a maintainer, I don't want to be in control of those credentials 😄 but perhaps @quinn-dougherty or @mdbecker would be good candidates to create the account so that it can be uploaded.

Closes https://github.com/CodeForPhilly/chime/issues/4

### CODE REVIEW

**Comment by nickcanz** [comment]
Further improvements would be to add a GitHub action to handle the publishing on every release via something like https://github.com/pypa/gh-action-pypi-publish but I didn't want to complicate this PR too much.

**Comment by lossyrob** [comment]
This looks great.

In PySTAC, we use travis to publish to pypi on tagged versions, which uses a token for auth information. This is a great setup in that publishing happens just by tagging:

https://github.com/azavea/pystac/blob/develop/.travis.yml#L16

Travis would have to be setup for the project (sign in with GitHub, add your repo) once a .travis.yml is built, and then a key for the encrypted pypi credentials:

https://docs.travis-ci.com/user/encryption-keys/#usage

I'm sure GitHub Actions can do this as well but I'm less familiar. Happy to help setup travis, although it's tough if you're not repo maintainer...one idea is that I could set up a .travis.yml and then jump on a video call with someone with ownership of the repo and we can work through it together.

**Comment by quinn-dougherty** [comment]
Closes https://github.com/CodeForPhilly/chime/issues/4

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    long_description_content_type="text/markdown",
+    url="https://github.com/CodeForPhilly/chime",
+    project_urls={
+        "Bug Reports": "https://github.com/CodeForPhilly/chime/issues",
+        "Source": "https://github.com/CodeForPhilly/chime",
+        "Documentation": "https://codeforphilly.github.io/chime/",
+    },
+    package_dir={'': 'src'},
+    packages=setuptools.find_namespace_packages(where='src', exclude=('tests')),
+    install_requires=REQUIREMENTS,
```
Comment (jlubken) [inline setup.py]: Please add the requirements here explicitly and put '-e .' in the requirements.txt file. Then the file read is no longer needed.

We'll need to test this with the deployments: `pip install .` without a requirements.txt file should be sufficient.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import setuptools
+
+from os import path
+
+with open("README.md", "r") as fh:
```
Comment (jlubken) [inline setup.py]: Use io.open here with the encoding.

**Comment by quinn-dougherty** [comment]
Good morning @nickcanz sorry about the merge conflict- it's a one-line change in https://github.com/CodeForPhilly/chime/pull/263

File: setup.py (hunk 2 above)
Comment (nickcanz) [reply setup.py]: Done!

File: setup.py (hunk 1 above)
Comment (nickcanz) [reply setup.py]: I put the requirements here explicitly, I'm not sure what you mean by `-e` in the requirements file, can you give me an example?

Comment (jlubken) [reply setup.py]: Thanks Nick. requirements.txt does not need the list of dependencies. Replacing the contents of requirements.txt with "-e ." will cause pip to find the list in setup.py. Kinda like delegation.

**Comment by jlubken** [comment]
Try using `pip install .`

**Comment by jlubken** [comment]
Add a COPY for setup.py and README.md to the dockerfile.

Remove the COPY requirements.txt from the dockerfile.

Change the install to either `pip install -q -e .` or `pip install -q .`

**Review by jlubken — APPROVED** [review]

**Pull request merged**